### DISCUSSION TITLE
Png-to-pdf's and webp-to-pdf's result size far different.

### DISCUSSION THREAD

**Discussion by z0cMWWKmGfudsDG** [ImageMagick/ImageMagick#5208] [Help] [answered]
I have two same pictures in different format. One is png (21.1 MiB), the other one is webp (13.2MiB). I use the command to test:
`convert test.png png.pdf`
`convert test.webp webp.pdf`
As a result, png.pdf's size is 26.1MiB, webp.pdf 'size is 142.7MiB.
How can be that? why webp.pdf is far bigger than png.pdf?

**Comment by fmw42** [comment, marked as answer]
PDF is just a vector shell on some image format. If the input png or web are 8-bit total color (palette) or compressed 24-bit color and the raster image that is put into the PDF shell is a 24-bit uncompressed color image, then the size will increase dramatically.

**Reply by z0cMWWKmGfudsDG** [reply]
Thank you.
When converting webp to pdf, can ImageMagick optimize the pdf just like converting png to pdf? And options for it?

**Reply by fmw42** [reply]
As far as I know, there are no optimize options for converting any format to PDF other than possibly compressions. But I am not an expert on PDF and what compressions or format is being used.

**Reply by fmw42** [reply]
See for example https://www.pdfa.org/pdfraster-an-overview/

**Comment by snibgo** [comment]
The difference is that a webp file has (in ImageMagick terms), "undefined" compression, whereas a png file has Zip compression. And those are the compressions used when writing the image in the PDF wrapper.

To get a smaller PDF file, use "-compress Zip" (or "-compress JPEG" or whatever you want) before the output filename.

**Reply by z0cMWWKmGfudsDG** [reply]
Thank you!
It is better, even better than the result of converting png to pdf.

**Comment by Jmuccigr** [comment]
Why not use a tool that puts images into PDFs with several options, namely [img2pdf](https://pypi.org/project/img2pdf/)? Use IM to get the original image to whatever size/quality you want, then simply insert it into the PDF with img2pdf
